File: 1527 B1. Palindrome Game (easy version)/Solution.cpp
// 314433734	      Apr/08/2025 07:22UTC+6	    Krishno_MD	      1527B1 - Palindrome Game (easy version)	      C++20 (GCC 13-64)	    Accepted	    109 ms	    100 KB

#include<bits/stdc++.h>
using namespace std;

//Krishno_MD_KUET_CSE

void solution(){
    int n; cin>>n;
    string str; cin>>str;
    int zero = 0;
    for(char i : str){
        if(i=='0'){
            zero++;
        }
    }
    if(zero==1 || zero%2==0){cout<<"BOB\n";}
    else{cout<<"ALICE\n";}
}

int main(){
	int t; cin>>t;
	for(int i=0; i<t; i++){solution();}
	return 0;
}

File: 1527 B1. Palindrome Game (easy version)/Solution.py
# 314433767	      Apr/08/2025 07:23UTC+6	      Krishno_MD	      1527B1 - Palindrome Game (easy version)	      PyPy 3-64	      Accepted	      156 ms	      4900 KB

#Krishno_MD_KUET_CSE

t = int(input())

for i in range(t):
    n = int(input())
    s = input()
    z = 0
    for i in s:
        if i=='0':
            z+=1
    if z==1 or z%2==0:
        print("BOB")
    else:
        print("ALICE")
